src/main/java/com/synopsys/integration/alert/provider/blackduck/tasks/BlackDuckAccumulator.java

```diff
@@ -99,12 +98,6 @@ public void accumulate() {
             saveNextSearchStart(nextSearchStartString);
         } catch (AlertDatabaseConstraintException e) {
             logger.error("Error occurred accumulating data! ", e);
-        } finally {
-            Optional<Long> nextRun = getMillisecondsToNextRun();
```

Comment:
Wouldn't it be better to fix the `getMillisecondsToNextRun()` method?

Reply:
Sounds like this will need to get fixed regardless of us logging the data. The method name now seems to be misleading.

Reply:
This was a discussion I had with Paulo so hopefully I can recall this correctly... The problem here is that when its run from the accumulator the ScheduledFuture object, inside of ScheduledTask, we are passing around is referring to the running accumulator job. This is a problem because when checking to see when the the job is done it will have already been done at the previous start time. We still want to keep that method in place because it is used in other places such as the taskManager which is working correctly to calculate the next run time.

Reply:
The issue here is that the task has a reference to the future that is a ScheduledFuture.  The ScheduledFuture api has a getDelay  method.  However we are asking the current running task for its delay as to when it will run next.  But the issue is that when we are asking the future for the delay it is returning a negative value because that is the delay when the task was started. It is the current running task so that is no positive delay to return.  For example -923 means this future was started 923 milliseconds ago because it is the current running task. This coincides with the logging message for when the task starts its run.  Negative numbers returned by getDelay means that the task is currently running or has run.  The delay has elapsed which is true.  

The TaskManager component works because the taskScheduler manages updating the future reference to wait to execute again after some time has passed.  The UI and anyone probing the TaskManager are asking for when a task next runs but it is outside the scope of the task.  Where we are logging is within the scope of the task which is running so getDelay will always be a negative millisecond value.  That is why the logging has been removed.  It doesn't actually provide any value at this point.